src/ui/web/src/composable/SynchronousWS.ts

```diff
@@ -0,0 +1,95 @@
+import { JsonArray } from '@/type/JsonType';
+import { MessageCode, ProtocolMessage, ProtocolRequest } from '@/type/UIProtocol';
+
+// let test = false;
+export default class SynchronousWS {
```

Comment:
WS are always synchronous, it's a WS client implementation proxying a bit the WebSocket class.